app/(candidate)/onboarding/[slug]/[step]/components/districts/DistrictTypeAutocomplete.js

```diff
@@ -0,0 +1,67 @@
+'use client'
+import React, { useEffect, useState } from 'react'
+import { Autocomplete } from '@mui/material'
+import TextField from '@shared/inputs/TextField'
+import { clientFetch } from 'gpApi/clientFetch'
+import { apiRoutes } from 'gpApi/routes'
+
+export default function DistrictTypeAutocomplete({
+  value,
+  onChange,
+  state,
+  electionYear,
+}) {
+  const [options, setOptions] = useState([])
+  const [loading, setLoading] = useState(false)
+
+  useEffect(() => {
+    let active = true
+
+    async function load() {
+      if (!state || !electionYear) return
+      setLoading(true)
+      try {
+        const { data = [] } = await clientFetch(
+          apiRoutes.elections.districts.types,
+          { state, electionYear },
+        )
+        if (active) {
+          setOptions(
+            data.map((d) => ({
+              ...d,
+              label: d.L2DistrictType.replace(/_/g, ' '),
+            })),
+          )
+        }
+      } finally {
+        active && setLoading(false)
+      }
+    }
```

Comment:
The actual fetching should be pulled out to an `async` method outside of the component, w/ the `try-catch` granually wrapping the `fetch` call.


Similar to what you did here:
https://github.com/thegoodparty/gp-webapp/pull/756/files#diff-91a133d95859de160d2e7688796a0db5ee4381a8cf5f7ad4dfdbe4b5066c30efR171-R185

<img width="670" height="353" alt="Image" src="https://github.com/user-attachments/assets/05a336aa-0b5e-4616-879c-c3ec65815130" />